lib/portlayer/exec/container.go

```diff
@@ -452,7 +452,11 @@ func (c *Container) Signal(ctx context.Context, num int64) error {
 	}
 
 	if num == int64(syscall.SIGKILL) {
-		return c.containerBase.kill(ctx)
+		err := c.containerBase.kill(ctx)
+		if err == nil {
+			publishContainerEvent(c.ExecConfig.ID, time.Now(), events.ContainerPoweredOff)
```

Comment:
Wouldn't this result in the powered off event getting published twice in some cases?

Reply:
There are three places where this `ContainerPoweredOff` event is used:

1. https://github.com/vmware/vic/blob/master/lib/portlayer/network/network.go#L122
If the container is already unbound by the previous `ContainerPowerOff` event, the second event would be a no-op.

2. https://github.com/vmware/vic/blob/master/lib/portlayer/exec/container.go#L647
This should not make any difference.

3. https://github.com/vmware/vic/blob/master/lib/apiservers/engine/backends/eventmonitor.go#L230
This would be a problem because `docker event` would show this event twice ... :(

Reply:
Regardless of where the event is currently being consumed, this would potentially put any future uses of the event in jeopardy. It also is not correct to fire the event twice for the same occurrence. I would rather we not do this fix, especially since it also does not really address the problem.